### PR TITLE
WIP: website

### PR DESCRIPTION
**Pull request by ceball**
https://ioam.github.io/geoviews/ is built on pushing to this PR.

To do:

* [x] Fill in placeholder content, making coherent
* [x] ensure haven't removed anything from existing site

Before merging:

* [ ] squash
* [x] update travis file to e.g. build site on tag only

### CODE REVIEW

**Comment by philippjfr** [comment]
Going to try to improve this PR a bit and then upload the build to geoviews.org. @ceball Did anything I need to know about change since this was last built?

**Comment by ceball** [comment]
Ah, sorry. I think what happened is I stopped being able to use geoviews on travis, and meant to come back to this PR a bit later (hoping the problem would resolve itself in the meantime), but then I forgot all about it. Sorry!

Do you know if the problem shown at https://travis-ci.org/ioam/geoviews/builds/314797817#L2101 has gone away? If so, I'd be happy to rebase this PR and get the build passing before you look at it.

**Comment by ceball** [comment]
Actually, I just checked, and it does seem to have gone away. So @philippjfr, just to confirm: I'll rebase and get things passing, then ping you, unless you've started already?

**Comment by philippjfr** [comment]
I haven't started yet, probably won't have time before Friday in any case.

**Comment by ceball** [comment]
@philippjfr, should be ok for you to continue with. nbsite hasn't had many of the improvements I'd hoped for yet, so there'll be some things that annoy/confuse you - please just ask me. Maybe you can motivate the next nbsite task I tackle. 

There are some usage notes for nbsite at https://ioam.github.io/nbsite/Usage.html, but they are still really draft notes.

**Comment by ceball** [comment]
Just a warning that a notebook failing to run isn't yet an error for nbsite - you have to notice! One notebook's currently failing: https://travis-ci.org/ioam/geoviews/jobs/327704526#L4581

**Comment by philippjfr** [comment]
I'm largely happy with this now, the two things I'd like to fix before merging are:

* Fix how latest_news.html is embedded, when empty it should not cause this (and it also needs to fixed when a twitter feed is embedded).

* I'd like to have the two-level nav bar back as described in https://github.com/ioam/sphinx_ioam_theme/issues/3.

**Comment by ceball** [comment]
> Fix how latest_news.html is embedded, when empty it should not cause weird wrapping (and it also needs to fixed when a twitter feed is embedded).

Sorry, right now, the only way to turn off the latest news (the twitter feed) is to completely remove it and its hacky outer div. I've done that (waiting for site to build to check it). We could surely do better. But before doing so, I'd like to decide which components of the sites are always going to be present, and how they might vary (https://github.com/ioam/sphinx_ioam_theme/issues/3). E.g. I think the twitter feed is a great way to have "latest news", but you might not necessarily need to have a separate feed for every project.

> before merging [...] I'd like to have the two-level nav bar back

I'd prefer to merge this PR first and address your request later on (but I'm not saying it's unimportant - the opposite!).

**Comment by philippjfr** [comment]
I'm happy to merge now, and address the nav bar issue once we've discussed it properly.

**Comment by philippjfr** [comment]
Any feedback/suggestions on the color scheme? Also is there a good place to record how to go about building the CSS with a new color scheme. My process is this:

1. Pick primary (dark) color (usually from the project logo)
2. Go to [Paletton](http://paletton.com/#uid=5310u0kj9nN8PBadWt0ogiGtOd6) and add the primary color, choosing a three color palette
3. Find the closest HTML color names for the three colors: http://chir.ag/projects/name-that-color/
4. Add the named colors to the variables.scss file: https://github.com/philippjfr/bokehplots.com/blob/master/theme/static/scss/base/_variables.scss
5. Replace the three colors throughout the scss files like in [this commit](https://github.com/philippjfr/bokehplots.com/commit/5f56baf25b4aa9218090d4e4613c15fe07c89fbe)
6. Compile the SCSS with compass and grunt

**Comment by ceball** [comment]
I like the color scheme/process for picking it. But I'm not a designer :)

I've added your useful comment to https://github.com/ioam/sphinx_ioam_theme/issues/1. I'll probably just add all documentation about how to customize stuff to the nbsite website (we'll see...).

**Comment by ceball** [comment]
The wrapping is fixed, so I think this PR is ready to be merged.

The docs will be re-built every commit, but ioam.github.io/geoviews site will be re-uploaded only when a tag is pushed. We'll come up with a better system, but I think that's ok for now?

**Comment by philippjfr** [comment]
>We'll come up with a better system, but I think that's ok for now?

Sounds fine. We may have to revisit once the site gets larger. I never did manage to improve the gallery scripts, but if I wanted to add a Gallery to GeoViews using the existing system, how much work would that be?

**Comment by ceball** [comment]
> if I wanted to add a Gallery to GeoViews using the existing system, how much work would that be

It shouldn't be too bad, and it was requested for datashader too. I think last time I tried it was working for mpl but not bokeh. I didn't investigate more at the time, but I'll try it out soon and get back to you if I need help...

**Comment by philippjfr** [comment]
Thanks for working on this! Looking good already and with some more tweaking this will be great.

File: .travis.yml [hunk 1]
```diff
@@ -1,32 +1,69 @@
...
+      python: "3.6"
+      script:
+        # TODO: can't remember why I did this twice
+        - python -c "import geoviews as gv; gv.sample_data('notebooks/user_guide/sample-data')"
+        - python -c "import geoviews as gv; gv.sample_data('doc/sample-data')"
+        - conda install -c conda-forge sphinx beautifulsoup4 graphviz
+        - pip install nbsite
+        - pip install sphinx_ioam_theme
+        # TODO: should make this content available too rather than deleting it!
+        - rm notebooks/*.ipynb
```
Comment (ceball) [inline .travis.yml]: @philippjfr, forgot to ask: what's in these notebooks, and should it appear somewhere on the website?

Comment (philippjfr) [reply .travis.yml]: No, I think they are old prototypes and can be deleted or updated. They almost certainly require updates.

**Pull request merged**